### ISSUE TITLE
global-color-identifiers-mode fails altogether, if it fails on some files

### ISSUE THREAD

**Issue by lockywolf** [ankurdave/color-identifiers-mode#82]
Hello, and thank you for your awesome package!

I found the following issue:

I have an org-mode file with the following fragment:

```
#+begin_src: elisp
(defun my-rtcite-export-handler (path desc format)
    "See http://www-public.imtbs-tsp.eu/~berger_o/weblog/2012/03/23/how-to-manage-and-export-bibliographic-notesrefs-in-org-mode/"
    (message "my-rtcite-export-handler is called : path = %s, desc = %s, format = %s" path desc format)
    (let* ((search (when (string-match "::#?\\(.+\\)\\'" path)
		     (match-string 1 path)))
	   []         (path (substring path 0 (match-beginning 0))))
      (cond ((eq format 'latex)
	     (if (or (not desc)
		     (equal 0 (search "rtcite:" desc)))
		 (format "\\cite{%s}" search)
	       (format "\\cite[%s]{%s}" desc search))))))
#+end_src
```

The code there is wrong, as the `[]` is there by mistake. However, `color-identifiers-mode` fails, and the `global-color-identifiers-mode` fails too.

**Comment by Hi-Angel** [comment]
Can't seem to reproduce it. What I did:

1. Commented out everything related to color-identifiers in my `.emacs`
2. Replaced my `color-identifiers-mode.el` with the upstream one *(I had some local changes to it)*
3. Opened `emacs`, removed everything from `*scratch*` buffer, did a M-x `org-mode`, copy-pasted your snippet
4. Did a M-x `global-color-identifiers-mode`. It got successfully enabled
5. Opened another file `test.c` and checked that it actually works and identifiers are getting highlighted

Is it perhaps fixed?
